app/services/validator_check_active_service.rb

```diff
@@ -0,0 +1,85 @@
+class ValidatorCheckActiveService
+
+  def initialize
+    # If stake lower of eqal this value validator becomes inactive
+    @stake_exclude_height = 0
+
+    # if delinquent for this period of time validator becomes inactive
+    @delinquent_time = 24.hours
+  end
+
+  def update_validator_activity
+    Validator.all.each do |validator|
+      if validator.scorable?
+        check_if_should_be_scorable(validator)
+      else
+        # Check validators that are currently not scorable in case they reactivate
+        if acceptable_stake?(validator) && not_delinquent?(validator)
+          validator.update(is_active: true)
+
+        # Check if vote account has been created for rpc_node
+        elsif validator.vote_accounts.exists?
+          validator.update(is_rpc: false) if validator.is_rpc
+        end
+      end
+    end
+  end
+  private
+
+  # number of previous by network
+  def previous_epoch(network)
+    current_epoch = EpochWallClock.where(network: network).order(created_at: :desc).last
+    if network == 'testnet'
+      @previous_epoch_testnet ||= current_epoch.epoch - 1
+    else
+      @previous_epoch_mainnet ||= current_epoch.epoch - 1
+    end
+  end
+
+  # returns true if validator has no history from previous epoch
+  def too_young?(validator)
+    !validator.validator_block_histories
+             .where(epoch: previous_epoch(validator.network))
+             .exists?
+  end
+
+  def check_if_should_be_scorable(validator)
```

Comment:
This method not only checks but also update objects, I think it should be splitted to smaller methods (see comment above in `update_validator_activity` method)